### ISSUE TITLE
[Feature Request] Change Date Format

### ISSUE THREAD

**Issue by mvpwizard** [keepassxreboot/keepassxc#2155]
Currently there is no way to change the date format from MM/dd/yyyy HH:mm. It would be better if the user could select from a list of predefined date formats as well as define their own.

## Expected Behavior
The user can change the date format in KeePassXC's settings

## Current Behavior
The user is restricted to the format MM/dd/yyyy HH:mm

## Context
I do not like the US date format because the month and day are out of order. This makes it hard to figure out when exactly a date is for me. I always prefer when the month's three letter code is used instead or when is is unambiguous which number represents the month. For example in yyyy-MM-dd the month number is unambiguously the middle number.

**Comment by TheZ3ro** [comment]
I cannot reproduce, see images below.
My guess is that Qt is guessing the date format from the language/operative system and probably you are using something like Snap or AppImage that fallback automatically to US format

![istantanea_2018-07-26_10-38-52](https://user-images.githubusercontent.com/686894/43251419-25a99d8a-90c0-11e8-87a7-4f37ea12cc5e.png)
![istantanea_2018-07-26_10-38-36](https://user-images.githubusercontent.com/686894/43251420-25d02acc-90c0-11e8-8b88-58d8ae1a29ce.png)

**Comment by mvpwizard** [comment]
Yes, I am using Snap. My operating system is using United Kingdom formats so I think that Snap that is responsible for the default to the US date format. I still think it would be a nice feature if I could set a custom date format though.

**Comment by droidmonkey** [comment]
I agree with the feature to set the date format within the app

**Comment by oittaa** [comment]
If the locale can't be detected, the default should be ISO 8601, not some random MM/dd/yyyy HH:mm.

**Comment by C0rn3j** [comment]
I'd also like ISO 8601 to be the default, no matter the locale.

While in some cases one could guess the format from the available data -
![image](https://user-images.githubusercontent.com/1641362/67789860-f4b28000-fa74-11e9-9ab3-b042949ebea9.png)

In others this is impossible.
![image](https://user-images.githubusercontent.com/1641362/67789889-0b58d700-fa75-11e9-8a18-78f56298b8d8.png)

There's no `yyyy-dd-mm` so ISO 8601 is something everyone understands, no matter whether they're from EU, USA or elsewhere.

**Comment by maricn** [comment]
+1 to the request..
can we make it read and respect `LC_TIME` envvar?

**Comment by aib** [comment]
In my case, it was Qt not playing nice with my locale settings (en_US + en_DK).

**Using `LC_TIME=en_SE` worked.**

<strike>
```
$ LC_TIME=en_DK.UTF-8 date +%c
2024-09-11T10:55:42 +03
$ LC_ALL=en_DK.UTF-8 LC_TIME=en_DK.UTF-8 ./release/KeePassXC.AppDir/usr/local/bin/keepassxc
```

Straight source build from current develop (c8fc25ea). Dates are still dd/mm/yyyy (or mm/dd/yyyy—I can't tell).

Whatever method is being used to detect locale is not working.

-- 

Seems to be due to Qt (CLDR?) not having en_DK. Setting LC_TIME to, say, en_CA changes the formats accordingly.

Now to find an english-speaking country which uses YYYY-mm-dd HH:MM:SS...

</strike>

**Comment by aib** [comment]
Seems to be due to Qt (CLDR?) not having en_DK. Setting LC_TIME to, say, en_CA changes the formats accordingly.

Now to find an english-speaking country which uses YYYY-mm-dd HH:MM:SS...

**Comment by acuteaangle** [comment]
+1 on this.
I would like the ability to set my date format, regardless of my system locale.
This could be a dropdown of common options, preferably also with a a strftime input.

I would like to use an ISO8601 datetime `YYYY-mm-ddTHH:MM:SS`. Many would prefer instead an ISO8601 date `YYYY-mm-dd` and time `HH:MM:SS` separated by a space.

As this is not standard for any locale, the only way for me to set this currently would be to build a custom locale.
I wish there was an easier way to customize global locale settings.

**Comment by abacao** [comment]
+1 I would like to use the US Language but the international measures of time in ISO8601. 

How can a feature like this be lacking after so many years? There are more users outside US.

Missing most in the "Expires" and Accessed/Modified fields

**Comment by johseg** [comment]
The workaround mentioned in https://github.com/keepassxreboot/keepassxc/issues/2155#issuecomment-2342925000 doesn't work for me unfortunately. I would also love to see the ability to have this configurable in keepassxc directly

**Comment by DMantis** [comment]
The workaround with manually redefined LC_TIME env var works with flatpak version.

**Comment by cherio** [comment]
I'm not sure if this is a new problem or a remnant of the existing one, but my time format seems to be completely unrelated to any of my settings. It is neither `ISO` / `ISO8601` nor it is based on my `LC_TIME="en_DK.UTF-8"`, nor it is `MM/dd/yyyy HH:mm`. This time format seems to be coming from nowhere in my system:

<img width="128" height="160" alt="Image" src="https://github.com/user-attachments/assets/0b2ab802-3730-45b5-896e-78675efd8c6f" />

The package (version 2.7.12) comes from the standard Arch Linux repository.

**Comment by AtomicFS** [comment]
From my experiments, the date-time format is very sketchy. Sometimes it does adhere to the `LANG` + `LC_TIME`, sometimes does not. But if it does, it is only partial - it always uses slash `/` as separator in date. Never hyphen `-`, even when `d_fmt="%Y-%m-%d"`.

```
$ LC_TIME=en_DK.UTF-8 LANG=en_DK.UTF-8 keepassxc
```

This is rather annoying. Either it should use the `LANG` + `LC_TIME` settings completely, or allow for in-application configuration.

Also, I am having difficulties to get the `LANG` + `LC_TIME` working as I have my own locale file definition in `/usr/share/i18n/locales/`, which KeepassXC is stubbornly trying to ignore apparently.

**Comment by CoelacanthusHex** [comment]
> From my experiments, the date-time format is very sketchy. Sometimes it does adhere to the `LANG` + `LC_TIME`, sometimes does not. But if it does, it is only partial - it always uses slash `/` as separator in date. Never hyphen `-`, even when `d_fmt="%Y-%m-%d"`.
> 
> ```
> $ LC_TIME=en_DK.UTF-8 LANG=en_DK.UTF-8 keepassxc
> ```
> 
> This is rather annoying. Either it should use the `LANG` + `LC_TIME` settings completely, or allow for in-application configuration.
> 
> Also, I am having difficulties to get the `LANG` + `LC_TIME` working as I have my own locale file definition in `/usr/share/i18n/locales/`, which KeepassXC is stubbornly trying to ignore apparently.

Qt doesn't use datetime format datebase from one bundled in glibc, instead, when build Qt, it generate datebase from CLDR database, and the CLDR database version is bind to Qt version (hardcode in build script). So you need to choose the locale which will use ISO 8601 in the CLDR database version bind to the Qt version which KeePassXC built against.
e.g. Qt 5.15 use v38.1, Qt 6.12 use v44.1. You can see CLDR version in comments of https://invent.kde.org/qt/qt/qtbase/-/blob/6.12/util/locale_database/cldr2qlocalexml.py
